packages/hardhat-core/test/internal/core/providers/http.ts

```diff
@@ -1,7 +1,37 @@
+import { assert } from "chai";
+import { MockAgent, MockPool } from "undici";
+
 import { HttpProvider } from "../../../../src/internal/core/providers/http";
+import { SuccessfulJsonRpcResponse } from "../../../../src/internal/util/jsonrpc";
 import { ALCHEMY_URL } from "../../../setup";
 
 describe("HttpProvider", function () {
+  const url = "http://some.node";
+  const networkName = "NetworkName";
+  const mockPool = (() => {
+    const agent = new MockAgent({
+      // as advised by https://undici.nodejs.org/#/docs/best-practices/writing-tests
+      keepAliveTimeout: 10, // milliseconds
+      keepAliveMaxTimeout: 10, // milliseconds
+    });
+    agent.disableNetConnect();
+    return new MockPool(url, { agent });
+  })();
+
+  describe("request()", function () {
+    it("should call mock pool's request()", async function () {
+      const response: SuccessfulJsonRpcResponse = {
+        jsonrpc: "2.0",
+        id: 1,
+        result: "whatever",
+      };
+      mockPool.intercept({ method: "POST", path: "/" }).reply(200, response);
+      const provider = new HttpProvider(url, networkName, {}, 20000, mockPool);
+      const result = await provider.request({ method: "net_version" });
+      assert.equal(result, response.result);
+    });
+  });
+
   describe("429 Too many requests - retries", function () {
```

Comment:
This test doesn't really work. We should replace it with a mocked one instead.

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/2414/commits/8e2ad09dcc67c590b9b836a283ff58ba6cdd7f33